libs/ngx-dsfr/navigation/src/lib/navigation.component.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Angular imports
+ */
+import { Component, ElementRef, Input, OnInit, QueryList, ViewChildren } from '@angular/core';
+import { Router } from '@angular/router';
+
+/**
+ * 3rd Party Imports
+ */
+import { UserRole } from '@betagouv/ngx-dsfr';
+
+/**
+ * TypeScript entities and constants
+ */
+export const EMPTY_NAVIGATION_ERROR: string =
+  'You MUST provide a navigation data for this component 😡 !!!';
+
+export interface NavigationItem {
+  id: string;
+  icon?: string;
+  label: string;
+  href: string;
+  isShow?: boolean;
+  userRoles?: UserRole[];
+  isChildAvailable?: boolean;
+  title?: string;
+  categories?: NavigationItem[];
+  description?: string;
+  child?: Pick<NavigationItem, 'title' | 'description'> & { items: NavigationItem[], isMega?: boolean }
+}
```

Comment:
I haven't seen them being used anywhere so I'm wondering about why do we have these properties : 
* `icon`
* `isShow`

If I remember correctly, you already explained why you added `userRoles`, but I'm guessing it will come as an improvement of this Component later...

Also, do we really need `isChildAvailable` ? Can't you replace it by a check on whether `childMenu` or `children` are undefined ?